source/scpd/types/Utils.d

```diff
@@ -31,14 +31,9 @@ public inout(opaque_vec!()) toVec (scope ref inout(Hash) data) nothrow @nogc
 /// Ditto
 public inout(opaque_vec!()) toVec (scope inout ubyte[] data) nothrow @nogc
 {
-    inout opaque_vec!() ret =
-    {
-        base: {
-            _start: data.ptr,
-            _end: data.ptr + data.length,
-            _end_of_storage: data.ptr + data.length,
-        },
-    };
+    inout opaque_vec!() ret;
+    foreach (elem; data)
+        ret.base.push_back(cast(ubyte) elem);
```

Comment:
This is a rather major change, as the previous version did not allocate

Reply:
Previous version was wrong tho, giving `std::vector` control of a GC allocated buffer seems dangerous.

Reply:
True, I thought there was a clever trick that was used that make this legit, but I can't find it (and everything I find confirms it was wrong). Can we at least call `reserve` before pushing bytes by bytes though ? :D